Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -748,6 +751,26 @@ void ComplexTextController::adjustGlyphsAndAdvances()
                     afterExpansion = false;
             }
 
+            const auto& textAutoSpace =  m_font.textAutospace();
+            float textAutoSpaceSpacing = 0;
+            if (textAutoSpace.hasIdeographAlpha()) {
+                char32_t previousCharacter = 0;
+                if (glyphIndex) {
+                    auto previousCharacterIndex = complexTextRun.indexAt(glyphIndex - 1);
+                    previousCharacter = static_cast<char32_t>(*(charactersPointer + previousCharacterIndex));
+                } else
+                    previousCharacter = m_textSpacingState.lastCharacterFromPreviousRun;
```

Comment:
We are in a loop here. Wouldn't it make more sense to just keep track of the previous character? Initialize it to `previousCharacter = m_textSpacingState.lastCharacterFromPreviousRun` before the start of the loop, and then update it as we go.

Reply:
We can do that, but then we need to instantiate `previousCharacter` even with `no-autospace`. I'm not sure what's better. I don't mind changing it.